### DISCUSSION TITLE
Does Prisma offer Transactions across multiple files?

### DISCUSSION THREAD

**Discussion by jaequery** [prisma/prisma#12974] [Q&A]
Coming from TypeORM, I am surprised to see that Prisma does not have any way to perform transactions in a similar way TypeORM does.

With TypeORM and the transactional-cls-hooked decorator, we can do such as:

```
@Transactional()
async updateManyThings() {
   await this.productService.updateStuff();
   await this.categoryService.updateAnotherStuff();
   await this.customerService.changeOtherStuff();
}
```

This allows so that if there is an error on any of the service calls, it'd safely rollback.

Does Prisma offer anything close to that? Only thing that I've come across is to perform the operation all inside a single method, which is not all that useful for us, as we rely heavily on using service files to perform mutable operations respectively.

**Comment by austincrim** [comment]
Hey @jaequery 👋 ,

This looks like a cool idea! We do not support a pattern like this yet. Would you mind opening a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with some more details?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
